client/lib/i18n-utils/utils.js

```diff
@@ -19,12 +19,25 @@ function getPathParts( path ) {
 }
 
 const i18nUtils = {
+	/**
+	 * Checks if provided locale is a default one.
+	 *
+	 * @param {string} locale - locale slug (eg: 'fr')
+	 * @return {boolean} true when the default locale is provided
+	 */
+	isDefaultLocale: function( locale ) {
+		return locale === config( 'i18n_default_locale_slug' );
+	},
```

Comment:
Not mandatory, but you could update [`switchLocale()`](https://github.com/Automattic/wp-calypso/blob/a00ceabc3944a1f8e67af017c3e6133e735c5619/client/lib/i18n-utils/switch-locale.js#L25) to use that new function now.